wrapper/src/main/java/software/amazon/jdbc/plugin/strategy/fastestresponse/FastestResponseStrategyPlugin.java

```diff
@@ -0,0 +1,199 @@
+
+  static {
+    PropertyDefinition.registerPluginProperties(FastestResponseStrategyPlugin.class);
+    PropertyDefinition.registerPluginProperties("frt-");
+  }
+
+  public FastestResponseStrategyPlugin(final PluginService pluginService, final @NonNull Properties properties) {
+    this(pluginService,
+        properties,
+        new HostResponseTimeServiceImpl(
+            pluginService,
+            properties,
+            RESPONSE_MEASURE_INTERVAL.getInteger(properties)));
+  }
+
+  public FastestResponseStrategyPlugin(
+      final PluginService pluginService,
+      final @NonNull Properties properties,
+      final @NonNull HostResponseTimeService hostResponseTimeService) {
+
+    this.pluginService = pluginService;
+    this.properties = properties;
+    this.hostResponseTimeService = hostResponseTimeService;
+    this.cacheExpirationNano = TimeUnit.MILLISECONDS.toNanos(RESPONSE_MEASURE_INTERVAL.getInteger(this.properties));
+  }
+
+  @Override
+  public Set<String> getSubscribedMethods() {
+    return subscribedMethods;
+  }
+
+  @Override
+  public Connection connect(
+      final String driverProtocol,
+      final HostSpec hostSpec,
+      final Properties props,
+      final boolean isInitialConnection,
+      final JdbcCallable<Connection, SQLException> connectFunc)
+      throws SQLException {
+
+    Connection conn = connectFunc.call();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return conn;
+  }
+
+  @Override
+  public Connection forceConnect(
+      final String driverProtocol,
+      final HostSpec hostSpec,
+      final Properties props,
+      final boolean isInitialConnection,
+      final JdbcCallable<Connection, SQLException> forceConnectFunc)
+      throws SQLException {
+
+    Connection conn = forceConnectFunc.call();
+    if (isInitialConnection) {
+      this.hostResponseTimeService.setHosts(this.pluginService.getHosts());
+    }
+    return conn;
+  }
+
+  @Override
+  public boolean acceptsStrategy(HostRole role, String strategy) {
+    return FASTEST_RESPONSE_STRATEGY_NAME.equalsIgnoreCase(strategy);
+  }
+
+  @Override
+  public HostSpec getHostSpecByStrategy(final HostRole role, final String strategy)
+      throws SQLException, UnsupportedOperationException {
+
+    if (!acceptsStrategy(role, strategy)) {
+      return null;
+    }
+
+    final HostSpec fastestResponseHost = latestResponseHost.get(role.name());
+    if (fastestResponseHost != null) {
+      HostSpec foundHostSpec = this.pluginService.getHosts().stream()
+          .filter(x -> x.equals(fastestResponseHost))
+          .findAny()
+          .orElse(null);
+
+      if (foundHostSpec != null) {
+        return foundHostSpec;
+      }
+    }
+
+    // Cached result isn't available. Need to find the fastest response time host.
+
+    final HostSpec calculatedFastestResponseHost = this.pluginService.getHosts().stream()
```

Comment:
fastestResponseHost ?

Reply:
"calculated" is important here since we discard a cached host and try to find the fastest host by looping through all hosts in the topology.